x-pack/server/lib/check_license/check_license.js

```diff
@@ -4,37 +4,33 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-export function checkLicense(xpackLicenseInfo) {
-  const pluginName = 'Index Management';
+import { RANKED_LICENSE_TYPES, LICENSE_STATUS } from '../constants';
+
+export function checkLicense(pluginName, minimumLicenseRequired, xpackLicenseInfo) {
+  if(!RANKED_LICENSE_TYPES.includes(minimumLicenseRequired)) {
+    throw new Error(`Invalid license type supplied to checkLicense: ${minimumLicenseRequired}`);
+  }
 
   // If, for some reason, we cannot get the license information
   // from Elasticsearch, assume worst case and disable
   if (!xpackLicenseInfo || !xpackLicenseInfo.isAvailable()) {
     return {
       isAvailable: false,
-      showLinks: true,
-      enableLinks: false,
+      status: LICENSE_STATUS.UNAVAILABLE,
       message: `You cannot use ${pluginName} because license information is not available at this time.`
     };
   }
 
-  const VALID_LICENSE_MODES = [
-    'trial',
-    'basic',
-    'standard',
-    'gold',
-    'platinum'
-  ];
-
-  const isLicenseModeValid = xpackLicenseInfo.license.isOneOf(VALID_LICENSE_MODES);
-  const isLicenseActive = xpackLicenseInfo.license.isActive();
-  const licenseType = xpackLicenseInfo.license.getType();
+  const { license } = xpackLicenseInfo;
+  const isLicenseModeValid = license.isOneOf([...RANKED_LICENSE_TYPES].splice(RANKED_LICENSE_TYPES.indexOf(minimumLicenseRequired)));
+  const isLicenseActive = license.isActive();
+  const licenseType = license.getType();
 
   // License is not valid
   if (!isLicenseModeValid) {
     return {
       isAvailable: false,
```

Comment:
What does `isAvailable` mean? Do we need it if we have the `UNAVAILABLE` status?